src/Api/HealthChecks/MailKitHealthCheck.cs

```diff
@@ -0,0 +1,39 @@
+using MailKit.Net.Smtp;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Passwordless.Common.Services.Mail;
+
+namespace Passwordless.Api.HealthChecks;
+
+public class MailKitHealthCheck : IHealthCheck
+{
+    private readonly MailKitSmtpMailProvider _mailProvider;
+
+    public MailKitHealthCheck(MailKitSmtpMailProvider mailProvider)
+    {
+        _mailProvider = mailProvider;
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = new CancellationToken())
+    {
+        SmtpClient client = null;
+        bool hasErrors = false;
+        try
+        {
+            client = await _mailProvider.GetClient();
+        }
+        catch
+        {
+            hasErrors = true;
```

Comment:
Can move the `return ...Unhealthy()` block here and the `finally` will still execute first